core/sql/generator/GenPreCode.cpp

```diff
@@ -4385,12 +4377,14 @@ RelExpr * GenericUpdate::preCodeGen(Generator * generator,
 	  GenExit();
 	}
 
-      if ((getOperatorType() == REL_HBASE_DELETE) &&
+      if ((getOperatorType() != REL_HBASE_UPDATE) &&
```

Comment:
Just curious. What value can getOperatorType() have at this point in code other than REL_HBASE_DELETE and REL_HBASE_UPDATE?